util.go

```diff
@@ -107,6 +107,24 @@ func escapeQuotes(s string) string {
 	return quoteEscaper.Replace(s)
 }
 
+func writeMultipartFormCustomData(w *multipart.Writer, params map[string]string, contentType string, r io.Reader) error {
```

Comment:
@Asker80 We are gonna improve it further.

1. Create a new unexported method called `createMultipartHeader`
```go
func createMultipartHeader(param, fileName, contentType string) textproto.MIMEHeader {
	hdr := make(textproto.MIMEHeader)
	hdr.Set("Content-Disposition", fmt.Sprintf(`form-data; name="%s"; filename="%s"`,
		escapeQuotes(param), escapeQuotes(fileName)))
	hdr.Set("Content-Type", contentType)
	return hdr
}
```
2. Change method signature to 
```go
func addMultipartFormField(w *multipart.Writer, mf *multipartField) error {
```
3. Call method `createMultipartHeader` from method `addMultipartFormField` to get part header

End result may look like-
```go
func addMultipartFormField(w *multipart.Writer, mf *multipartField) error {
	partWriter, err := w.CreatePart(createMultipartHeader(mf.Param, mf.FileName, mf.ContentType))
	if err != nil {
		return err
	}

	_, err = io.Copy(partWriter, r)
	return err
}
```

Optional:

Call `createMultipartHeader` from method `writeMultipartFormFile`. Otherwise I will take care after your PR merge.